### PR TITLE
Add api to serialize and deserialize chunk to bytes

### PR DESCRIPTION
**Pull request by REDxEYE**
Simple API to serialize chunk to LuaBytearray and deserialize it back

### CODE REVIEW

File: src/logic/scripting/lua/libs/libworld.cpp [hunk 1]
```diff
@@ -115,6 +119,76 @@ static int l_get_generator(lua::State* L) {
     return lua::pushstring(L, require_world_info().generator);
 }
 
+static int l_get_chunk_data(lua::State* L) {
+    int x = (int)lua::tointeger(L, 1);
+    int y = (int)lua::tointeger(L, 2);
+    const auto& chunk = level->chunks->getChunk(x, y);
+    if (chunk->isEmpty()) {
```
Comment (MihailRis) [inline src/logic/scripting/lua/libs/libworld.cpp]: potential nullptr dereference

File: src/logic/scripting/lua/libs/libworld.cpp [hunk 2]
```diff
@@ -115,6 +119,76 @@ static int l_get_generator(lua::State* L) {
...
+        return 0;
+    }
+
+    bool compress = false;
+    if (lua::gettop(L) >= 3) {
+        compress = lua::toboolean(L, 3);
+    }
+
+    if (compress) {
+        return lua::newuserdata<lua::LuaBytearray>(
```
Comment (MihailRis) [inline src/logic/scripting/lua/libs/libworld.cpp]: tests i've made for my implementation show that gzip over extrle16 is the best combination for chunks:

| average        | gzip | extrle16 | extrle16 + gzip |
| -------------- | ---- | -------- | --------------- |
| time (mcs)     | 1196 | 78       | 176             |
| output (bytes) | 1476 | 2642     | 1011            |

File: src/logic/scripting/lua/libs/libworld.cpp (hunk 1 above)
Comment (REDxEYE) [reply src/logic/scripting/lua/libs/libworld.cpp]: Yes, i noticed that isEmpty is a Chunk method, not sharedptr

File: src/logic/scripting/lua/libs/libworld.cpp (hunk 2 above)
Comment (REDxEYE) [reply src/logic/scripting/lua/libs/libworld.cpp]: Good to know, thank you

File: CMakeLists.txt [hunk 3]
```diff
@@ -82,3 +82,11 @@ if (VOXELENGINE_BUILD_TESTS)
     enable_testing()
     add_subdirectory(test)
 endif()
+
+add_custom_target(copy_resources ALL
```
Comment (MihailRis) [inline CMakeLists.txt]: Line 79?

Comment (REDxEYE) [reply CMakeLists.txt]: Whoops, line 79 wast copying resources for me so i added my own version of it. I'll remove that

Comment (REDxEYE) [reply CMakeLists.txt]: Fixed

**Pull request merged**